content/blog/2020/05/citrix-policy-migration.md

```diff
@@ -0,0 +1,167 @@
+
+# Export
+1.	Download the Citrix Migration Tool from the xenapp 7.6 download page: [Citrix 7.6 Downloads](https://www.citrix.com/downloads/xenapp/product-software/xenapp-76-enterprise-edition.html)
+     * Important note, be sure to download the migration "scripts", not the migration "tool". The files we need are ReadIMA.zip and ImportFMA.zip. If you see XDExport.zip and XDImport.zip, you have the the actual tool and it will not work. 
+	 ![migration_script_download](/img/2020/05/migration_script_download.png)
+2.	On a 6.5 farm server ensure you have windows management framework 3 or above installed, I used WMF 4. You can download here: [Microsoft WMF 4.0](https://www.microsoft.com/en-us/download/details.aspx?id=40855)
+3.	Extract ReadIMA.zip to c:\temp
+4.	Open and administrator command prompt and execute the following:
+     * Note if you get security warnings press "R" to ignore and load scripts
+     ![script_warning](/img/2020/05/script_warning.png)
+
+```bash
+cd c:\temp\ReadIMA
+import-module .\exportpolicy.psd1
+import-module .\exportxafarm.psd1
+export-policy -XmlOutputFile .\policies.xml -LogFile .\policy-export.log
+```
+
+5.	After the policy export completes, copy the policies.xml file from the 6.5 server to a 7.xx delivery controller and place in c:\temp\ImportFMA
+6.	If you have a large number of policies you will want to generate a verification file now. You can use to your compare current policies the to the new farm later on. On your 6.5 server run the below commands: 
+
+```bash
+if (!(Get-PSDrive | where name -eq LocalFarmGpo)) { 
+asnp citrix*
+import-module .\exportpolicy.psd1
+import-module .\exportxafarm.psd1
+}
+cd LocalFarmGpo:\
+$userpolicies=Get-ChildItem .\User |Select-Object name
+$outfile="c:\temp\ctxpolicy-old.txt"
+foreach ($p in $userpolicies) {
+    Get-ItemProperty -path .\User\$($p.name)  |Out-File $outfile -Append
+    Get-ChildItem .\User\$($p.name) -Recurse |where state -eq enabled |Out-File $outfile -Append
+    Get-ChildItem .\User\$($p.name) -Recurse |where enabled -eq true |Out-File $outfile -Append
+}
+```
+
+7.	This will create a file c:\temp\ctxpolicy.txt (feel free to rename if desired) that contains all of policies on the farm. Save this for the verification process later. 
+
+
+# Import
```

Comment:
general question, why do we have to edit this powershell code? the reader might want to know as well

Reply:
added explanation below #import . 2 reasons. 1. [System.Reflection.Assembly]::LoadWithPartialName is depreciated and 2. Citrix changed the name for the policy management snapin. Also updated code in step 4 to use "get-pssnapin" rather than the alias, "asnp" so it matches my explanation.